src/java/org/apache/cassandra/service/accord/async/AsyncOperation.java

```diff
@@ -179,7 +179,8 @@ private void finish(R result)
     private void fail(Throwable throwable)
     {
         Invariants.nonNull(throwable);
-        Invariants.checkArgument(state != State.FINISHED && state != State.FAILED, "Unexpected state %s", state);
+        if (!(state != State.FINISHED && state != State.FAILED))
```

Comment:
```suggestion
        if (state == State.FINISHED || state == State.FAILED)
```

Reply:
I pushed the knowledge of complete to the `State` via a `isComplete()` method, and switched to `if (state.isComplete())`